<a href="https://colab.research.google.com/github/randiijulian/Zenius-ML/blob/main/Day10HW_Randi_Julian_Saputra.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Day 10 Assignment

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


import library yang dibutuhkan

In [3]:
import scipy.stats as stats

A company wants to run a new campaign to increase the transactions’ amount of its product.
Before running the actual campaign to the whole customers, the company wants to know the
campaign’s effectiveness in their 30 days transactions’ amount.
The average of transactions’ amount before the campaign is $100.
The company ran a testing on 20 customers and gathered their transactions’ amount in the next 30
days after the campaign. The data is stored in this table below.
Run a t-test analysis on the transactions’ amount average (with alpha = 5%).

In [4]:
data = [100, 150, 50, 100, 130, 120, 100, 110, 75, 65, 150, 120, 50, 100, 100, 140, 90, 150, 50, 90]

## Define H0 and H1

H0 : The average of transactions after the campaign = 100

- One tailed
    - H1 : The average of transactions after the campaign > 100
    - H1 : The average of transactions after the campaign < 100


- Two tailed
    - H1 : The average of transactions after the campaign != 100

In [5]:
# perform one sample t-test
stats.ttest_1samp(a=data, popmean=100)

Ttest_1sampResult(statistic=0.2731657711666385, pvalue=0.787672894169515)

* If using alpha = 0.05, then the concolusion is P-value larger than alpha, meaning we fail to reject H0
* Conclusion: The mean is equal to 100
* If using t statistic, we have to find the critical region first:

In [6]:
# For two tailed test:
stats.t.ppf(1-(0.05/2), 19)

2.093024054408263

* Critical Region: t < - 2.09 or t > 2.09
* t statistic = 0.2731657711666385
* As t statistic is not in the critical region, then we fail to reject H0
* Conclusion: The mean is equal to 100

In [7]:
# For One tailed test:
0.787672894169515/2

0.3938364470847575

* If using alpha = 0.05, use P-value/2 = 0.39. The concolusion is P-value/2 larger than alpha, meaning we fail to reject H0
* Conclusion: The mean is equal to 100

* If using t statistic, we have to find the critical region first:

In [8]:
# If H1: mean > 100 (right side)
stats.t.ppf(1-0.05, 19)

1.729132811521367

* Critical Region: t > 1.72
* t statistic = 0.2731657711666385
* As t statistic is not in the critical region, then we fail to reject H0
* Conclusion: The mean is equal to 100

In [9]:
# If H1: mean < 100 (left side)
stats.t.ppf(0.05, 19)

-1.7291328115213678

* Critical Region: t < -1.72
* As t statistic is not in the critical region, then we fail to reject H0
* Conclusion: The mean is equal to 100